File: 华为od2026题库/2026年6月28日/P14410/data/foo.js
var countIsolatedIntervals = function(intervals) {
    var n = intervals.length;
    var ans = 0;
    for (var i = 0; i < n; i++) {
        var s1 = intervals[i][0], e1 = intervals[i][1];
        var isolated = true;
        for (var j = 0; j < n; j++) {
            if (i === j) continue;
            var s2 = intervals[j][0], e2 = intervals[j][1];
            // 标准区间相交判定
            if (s1 <= e2 && s2 <= e1) {
                isolated = false;
                break;
            }
        }
        if (isolated) ans++;
    }
    return ans;
};

File: 华为od2026题库/2026年6月28日/P14410/data/foo.c
#include <stdbool.h>

int countIsolatedIntervals(int** intervals, int intervalsSize, int* intervalsColSize) {
    int ans = 0;
    for (int i = 0; i < intervalsSize; i++) {
        int s1 = intervals[i][0], e1 = intervals[i][1];
        bool isolated = true;
        for (int j = 0; j < intervalsSize; j++) {
            if (i == j) continue;
            int s2 = intervals[j][0], e2 = intervals[j][1];
            // 标准区间相交判定
            if (s1 <= e2 && s2 <= e1) {
                isolated = false;
                break;
            }
        }
        if (isolated) ans++;
    }
    return ans;
}
